### PR TITLE
CLI module create behavior with existing meta.jsons

### PR DESCRIPTION
**Pull request by zaporter-work**
I was talking with Hazal during the happy-hour and she was voicing how it was unintuitive that she had to both create a meta.json (as the docs and other modules say to do) and yet also not create it until she runs `viam module create`.

This PR updates that behavior so that if a meta.json exists in the current directory (with a matching moduleid as what was requested via the cli args), it will create the module (in the registry) and then update that module (reducing the number of steps by 1)

It maintains the existing behavior for directories where there is no meta.json.

User flow:
1) New user wants to create a module
2) User copies files from some online examples (including a meta.json)
3) They start editing files / get their module to a semi-working state as a local module
4) They edit the meta.json to have the info they expect for their module (including a new module id) (ex `module_id:"abe:mycoolmodule"`)
5) They look online and try to figure out what to do from here. They probably look into `viam module update` or `viam module create`.
5) They run `viam module create --name mycoolmodule --public-namespace abe`
6) Instead of printing an error message (current behavior), the cli just runs the create and update flows (proposed new behavior).

Alternatives / also considered:
`viam module update` -> detect if the module does not exist, create it if it doesn't, then run update.
This /could/ help, but it could also result in lots of user confusion with lots of new modules (when people /think/ that they are renaming their module but they are actually creating new modules)

`viam module create (no args)` -> use name /namespace from meta.json. I also think this will cause user confusion if they copy the meta.json without modifying it.

EDIT:
`viam module create (no args)` -> begin interactive walkthrough where user can enter/edit both the public namespace + name of the module (and maybe enter things like visibility, description, url, etc) (would require more thought)

### CODE REVIEW

File: cli/module_registry.go
Comment (zaporter-work) [inline cli/module_registry.go:452]: Flyby: I figured out how to replicate Alexa's issue if you manually create the tarball instead of the auto-bundling.

 ```
Error validating module: the module archive contains a directory at the entrypoint "testdir" instead of an executable file. For more details, please visit: https://docs.viam.com/fleet/cli/#module
```

I can break this out into another PR if you want.

File: cli/module_registry.go
Comment (zaporter-work) [inline cli/module_registry.go:136]: could also silently fail (or just not update... but then people /will/ be confused about why the info isn't on the webpage they were told to visit and which is in the meta.json file for the module they just created.)

File: cli/module_registry.go
Comment (abe-winter) [inline cli/module_registry.go:105]: you're ignoring the `err != nil` case because if the file is missing or invalid, you're going to overwrite it

that is correct for the 'missing' case, probably okay in the 'invalid' case

fwiw it took me two reads to understand why you were targeting `err == nil` case

File: cli/module_registry.go
Comment (abe-winter) [inline cli/module_registry.go:110]: not totally sure what the user hint should be here

from the workflow you gave in the PR description, they might be trying to adapt a sample meta.json, so 'delete it' is not always the right call

guessing you have a better understanding of the user flow here than I do, so I'm just flagging this, your call what to do

**Review by abe-winter — APPROVED** [review]
this looks okay code-wise so I am :heavy_check_mark: 

consider rewriting the error message on L110 in light of new user flow

your `viam module create (no args)` idea is also good. I'm on the fence from a UX perspective. 'CLI args must agree with meta.json' makes sense, but raises the question 'why do I need both'. taking module_id from meta.json makes sense, but raises the issue of blindly using a meta.json you downloaded from docs (though worst case there is getting rejected by backend). I think I don't have a clear preference here, am fine with what you chose, but would wait for a second review here

File: cli/module_registry.go
Comment (zaporter-work) [reply cli/module_registry.go:105]: The err!=nil case has a nil manifest which is checked later.

Agreed that this code is very hard to read.

I'll refactor to split into two separate code paths if we end up liking this user flow.

**Comment by abe-winter** [comment]
on second thought -- if people ask for a 'no args' version later, we can add it. choosing the 'args must match' approach today doesn't lock us in going forward

File: cli/module_registry.go
Comment (zaporter-work) [reply cli/module_registry.go:110]: Agreed -- it is a weird hint. That was the error message from before (when we didn't support this edge case args-match case). 

The reason I proposed that they delete it is that they are guaranteed to arrive at a working state if they delete and recreate instead of telling them about how to modify the manifest module_id which could be error prone and might lead to more frustration than just recreating the file.

Thoughts?

Comment (abe-winter) [reply cli/module_registry.go:110]: works for me

File: cli/module_registry.go
Comment (michaellee1019) [reply cli/module_registry.go:105]: Agreed, refactor to do` err != nil `and then `else` to match golang conventions.

File: cli/module_registry.go
Comment (michaellee1019) [reply cli/module_registry.go:110]: May I suggest: "A different module '%q' exists in the meta.json of the current directory. Either delete the meta.json or modify the name, and try again."

File: cli/module_registry.go
Comment (michaellee1019) [reply cli/module_registry.go:136]: This is fine by me. I'm curious to see what users will naturally encounter. I guess if they clear out descriptions, models, etc.

**Review by michaellee1019 — COMMENTED** [review]
TY for addressing feedback so quickly! The user flow makes sense to me. Pretty sure i did this once too.

**Review by michaellee1019 — APPROVED** [review]

**Pull request merged**
